diff --git a/lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.cpp b/lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.cpp
index 7c42451a8df8..2d6cd901606c 100644
--- a/lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.cpp
+++ b/lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.cpp
@@ -1754,7 +1754,7 @@ size_t ObjectFileELF::GetSectionHeaderInfo(SectionHeaderColl &section_headers,
         const uint64_t section_size =
             sheader.sh_type == SHT_NOBITS ? 0 : sheader.sh_size;
         llvm::StringRef name(shstr_data.PeekCStr(I->sh_name));
-        I->section_name = name.str();
+        I->section_name = ConstString(name);
 
         if (arch_spec.IsMIPS()) {
           uint32_t arch_flags = arch_spec.GetFlags();
@@ -1917,7 +1917,7 @@ lldb::user_id_t ObjectFileELF::GetSectionIndexByName(llvm::StringRef name) {
   if (name.empty() || !ParseSectionHeaders())
     return 0;
   for (size_t i = 1; i < m_section_headers.size(); ++i)
-    if (m_section_headers[i].section_name == name)
+    if (m_section_headers[i].section_name.GetStringRef() == name)
       return i;
   return 0;
 }
@@ -1961,7 +1961,7 @@ SectionType ObjectFileELF::GetSectionType(const ELFSectionHeaderInfo &H) const {
   case SHT_DYNAMIC:
     return eSectionTypeELFDynamicLinkInfo;
   }
-  return GetSectionTypeFromName(H.section_name);
+  return GetSectionTypeFromName(H.section_name.GetStringRef());
 }
 
 static Permissions GetPermissions(const ELFSectionHeader &H) {
@@ -2167,7 +2167,7 @@ void ObjectFileELF::CreateSections(SectionList &unified_section_list) {
        I != m_section_headers.end(); ++I) {
     const ELFSectionHeaderInfo &header = *I;
 
-    const std::string &name = I->section_name;
+    llvm::StringRef name = I->section_name.GetStringRef();
     const uint64_t file_size =
         header.sh_type == SHT_NOBITS ? 0 : header.sh_size;
 
@@ -2187,7 +2187,7 @@ void ObjectFileELF::CreateSections(SectionList &unified_section_list) {
         this,            // ObjectFile to which this section belongs and should
                          // read section data from.
         SectionIndex(I), // Section ID.
-        name,            // Section name.
+        name.str(),      // Section name.
         sect_type,       // Section type.
         InfoOr->Range.GetRangeBase(), // VM address.
         InfoOr->Range.GetByteSize(),  // VM size in bytes of this section.
@@ -3423,7 +3423,7 @@ void ObjectFileELF::RelocateSection(lldb_private::Section *section)
   for (SectionHeaderCollIter I = m_section_headers.begin();
        I != m_section_headers.end(); ++I) {
     if (I->sh_type == SHT_RELA || I->sh_type == SHT_REL) {
-      llvm::StringRef hay_name(I->section_name);
+      llvm::StringRef hay_name = I->section_name.GetStringRef();
       if (hay_name.empty())
         continue;
       if (needle == hay_name || needlea == hay_name) {
@@ -3770,7 +3770,7 @@ void ObjectFileELF::DumpELFSectionHeaders(Stream *s) {
        I != m_section_headers.end(); ++I, ++idx) {
     s->Printf("[%2u] ", idx);
     ObjectFileELF::DumpELFSectionHeader(s, *I);
-    const std::string &section_name = I->section_name;
+    llvm::StringRef section_name = I->section_name.GetStringRef();
     if (!section_name.empty())
       *s << ' ' << section_name << "\n";
   }
diff --git a/lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.h b/lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.h
index 8adb8af10fe1..977651ae8a9f 100644
--- a/lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.h
+++ b/lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.h
@@ -182,7 +182,7 @@ private:
   typedef std::vector<elf::ELFProgramHeader> ProgramHeaderColl;
 
   struct ELFSectionHeaderInfo : public elf::ELFSectionHeader {
-    std::string section_name;
+    lldb_private::ConstString section_name;
   };
 
   typedef std::vector<ELFSectionHeaderInfo> SectionHeaderColl;
