diff --git a/utils/bazel/llvm-project-overlay/clang/BUILD.bazel b/utils/bazel/llvm-project-overlay/clang/BUILD.bazel
index f3ad2b7e21b4..9d36321d56d2 100644
--- a/utils/bazel/llvm-project-overlay/clang/BUILD.bazel
+++ b/utils/bazel/llvm-project-overlay/clang/BUILD.bazel
@@ -273,6 +273,13 @@ td_library(
     deps = [":BuiltinsBaseTdFiles"],
 )
 
+td_library(
+    name = "BuiltinsARMBaseTdFiles",
+    srcs = ["include/clang/Basic/BuiltinsARMBase.td"],
+    includes = ["include"],
+    deps = [":BuiltinsBaseTdFiles"],
+)
+
 td_library(
     name = "BuiltinsAMDGPUDocsTdFiles",
     srcs = ["include/clang/Basic/BuiltinsAMDGPUDocs.td"],
@@ -331,6 +338,17 @@ gentbl_cc_library(
     ],
 )
 
+gentbl_cc_library(
+    name = "basic_builtins_arm_gen",
+    tbl_outs = {"include/clang/Basic/BuiltinsARM.inc": ["-gen-clang-builtins"]},
+    tblgen = ":clang-tblgen",
+    td_file = "include/clang/Basic/BuiltinsARM.td",
+    deps = [
+        ":BuiltinsARMBaseTdFiles",
+        ":BuiltinsBaseTdFiles",
+    ],
+)
+
 gentbl_cc_library(
     name = "basic_builtins_bpf_gen",
     tbl_outs = {"include/clang/Basic/BuiltinsBPF.inc": ["-gen-clang-builtins"]},
@@ -767,6 +785,7 @@ cc_library(
         ":basic_builtin_templates_gen",
         ":basic_builtins_aarch64_gen",
         ":basic_builtins_amdgpu_gen",
+        ":basic_builtins_arm_gen",
         ":basic_builtins_bpf_gen",
         ":basic_builtins_directx_gen",
         ":basic_builtins_gen",
