diff --git a/utils/bazel/llvm-project-overlay/llvm/BUILD.bazel b/utils/bazel/llvm-project-overlay/llvm/BUILD.bazel
index 940998fea026..0210328f739b 100644
--- a/utils/bazel/llvm-project-overlay/llvm/BUILD.bazel
+++ b/utils/bazel/llvm-project-overlay/llvm/BUILD.bazel
@@ -4440,8 +4440,11 @@ cc_library(
         "lib/ExecutionEngine/Orc/Shared/*.cpp",
     ]),
     hdrs = glob([
+        "include/llvm/ExecutionEngine/Orc/*.h",
         "include/llvm/ExecutionEngine/Orc/Shared/*.h",
         "include/llvm/ExecutionEngine/Orc/RTBridge/*.h",
+        "include/llvm/ExecutionEngine/Orc/RTBridge/SPS/**/*.h",
+        "include/llvm/ExecutionEngine/JITLink/*.h",
     ] + [
         "include/llvm/ExecutionEngine/Orc/SymbolStringPool.h",
     ]),
